task/upload.go

```diff
@@ -72,27 +74,57 @@ func handleUploadVOD(p handleUploadVODParams) (*TaskHandlerOutput, error) {
 		if err != nil {
 			return nil, err
 		}
-		var encryption *clients.EncryptionPayload
-		params := tctx.Task.Params.Upload
 
-		if params != nil && params.Encryption.EncryptedKey != "" {
-			encryption = &clients.EncryptionPayload{
-				EncryptedKey: params.Encryption.EncryptedKey,
+		var encryption *clients.EncryptionPayload
+		var clipStrategy *video.ClipStrategy
+
+		if tctx.Task.Params.Clip != nil {
+			clipParams := tctx.Task.Params.Clip
+			if clipParams != nil {
+				clipStrategy = &video.ClipStrategy{
+					StartTime:  clipParams.ClipStrategy.StartTime,
+					EndTime:    clipParams.ClipStrategy.EndTime,
+					PlaybackID: clipParams.ClipStrategy.PlaybackId,
+				}
```

Comment:
Isn't this the exact same type+object? Why do we need to recreate it? Also on `TaskClip`